### ISSUE TITLE
How to do prediction?

### ISSUE THREAD

**Issue by jcthink** [liznerski/fcdd#15]
Hi 
Thanks for your great work.
Could you please explain how to detect and localize  anomaly or generate a heatmap for the test image with the .pt file generated?

**Comment by liznerski** [comment]
Hey. I recommend using the [custom dataset runner](https://github.com/liznerski/fcdd#custom-data). Just put at least one image in `custom/test/classX/normal`, `custom/test/classX/anomalous`, and `custom/train/classX/normal`. You can also put the same image in all folders in case you want to use just one. Then, run `python runners/run_custom.py --supervise-mode unsupervised --it 1 -e 0 --load MY_SNAPSHOT.pt`.  You can find the generated heatmaps in the log directory. 

If you haven't used the VGG net, you need to change the architecture with `-net`. Note that, if your architecture expects inputs of different shape than (3, 224, 224), you need to modify the transforms [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/datasets/image_folder.py#L81). In fact, for the best results you could change the transforms to use the mean and std of your training data. 

How did you train your model? You should get heatmaps for images of the test set after having trained the model. Isn't that sufficient already?

**Comment by liznerski** [comment]
Another comment on this. If you want the raw pixel values of the heatmaps that are generated, you can find these in the "tims" directory and load them via `torch.load("heatmaps_paper_global_lbl0.pth")`.
If you want a script that loads the snapshot of a model and generates pixel-wise anomaly score tensors, I added such an implementation [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/runners/run_prediction_with_snapshot.py). 
I hope this helps.

**Comment by opentld** [comment]
> Another comment on this. If you want the raw pixel values of the heatmaps that are generated, you can find these in the "tims" directory and load them via `torch.load("heatmaps_paper_global_lbl0.pth")`. If you want a script that loads the snapshot of a model and generates pixel-wise anomaly score tensors, I added such an implementation [here](https://github.com/liznerski/fcdd/blob/master/python/fcdd/runners/run_prediction_with_snapshot.py). I hope this helps.

I have run run_prediction_with_snapshot.py,  and get the all_anomaly_scores tensor，but how to judge whether it is abnormal through pixel-wise anomaly score tensors?  and how to get the heat map through the run_prediction_with_snapshot.py method?

@liznerski

**Comment by opentld** [comment]
another question：
I only put one picture,and the path is shown in the figure:
![image](https://user-images.githubusercontent.com/21096515/139523266-cfe52439-6029-4f73-878a-c0a822bbb843.png)
![004](https://user-images.githubusercontent.com/21096515/139523480-0faad1e0-97ca-4c7b-a998-d586b5d5e22d.png)
 then I ran python runners/run_custom.py --supervise-mode unsupervised --it 1 -e 0 --load ../../data/test/model/snapshot.pt --net FCDD_CNN224_VGG_F
Get the following result：
![image](https://user-images.githubusercontent.com/21096515/139523344-e2bc3460-f39f-4526-96f0-c7ecaad2ebe5.png)
![image](https://user-images.githubusercontent.com/21096515/139523358-043ac6e8-09b3-41c1-b68b-6ebbf869b109.png)
heatmaps_global.png as shown:
![heatmaps_global](https://user-images.githubusercontent.com/21096515/139523378-f837021b-aa86-4bfc-bf3c-b63a75d9c68e.png)
heatmaps_paper_global_lbl0.png:
![heatmaps_paper_global_lbl0](https://user-images.githubusercontent.com/21096515/139523421-3ac1f18d-60e6-42c1-b2de-fd38c9052693.png)
heatmaps_paper_global_lbl1.png:
![heatmaps_paper_global_lbl1](https://user-images.githubusercontent.com/21096515/139523427-a69069ad-c1b5-4c73-9cff-f8ffbc8e9e15.png)
heatmaps_paper_local_lbl0.png:
![heatmaps_paper_local_lbl0](https://user-images.githubusercontent.com/21096515/139523434-17cbb0d0-b3d5-4af1-ab1d-7e0ec25c236e.png)
heatmaps_paper_local_lbl1.png:
![heatmaps_paper_local_lbl1](https://user-images.githubusercontent.com/21096515/139523442-563454a3-ce9f-4e86-97c1-f42ef7835b45.png)
train_heatmaps_global.png:
![train_heatmaps_global](https://user-images.githubusercontent.com/21096515/139523448-202cb5d4-eaa5-4195-8620-b7aeb1812dfc.png)

so, why does run_custom.py try to process all the classes? For example, I did not place the cable picture, but it still generates normal_1? I just want to perform anomaly detection on the bottle image I set, and get its classification result and heat map. How can I do it?

**Comment by liznerski** [comment]
Commented in #22.
